cmd_manager.py

```diff
@@ -160,25 +165,69 @@ async def shop(self, ctx):
                 await shop_msg.add_reaction(arrow)
 
     @disc_cmds.command(name='timer')
-    async def timer(self, ctx, duration: int, unit='sec', name='', *args):
-        if len(args) > 0:
-            await ctx.send(content='Invalid number of arguments, please try again.')
-            return None
+    async def timer(self, ctx, *args):
+        message = ' '.join(args)
 
-        sec_duration = duration
-        if unit in ['min', 'minute', 'minutes', 'm']:
-            sec_duration = duration * 60
-        elif unit in ['hr', 'hour', 'hours', 'h']:
-            sec_duration = duration * 3600
-        elif unit in ['day', 'days', 'd']:
-            sec_duration = duration * 86400
-        elif unit not in ['sec', 'second', 'seconds', 's']:
-            await ctx.send(content='Invalid time unit specified, please try again.')
+        # Create a Calendar instance for time parsing
+        cal = parsedatetime.Calendar()
+
+        # Parse the message and obtain time delta
+        curr_time = datetime.now()
+        # res format: (datetime, result_code, start_idx, end_idx, matched_string)
+        res = cal.nlp(message)
+
+        # If the nlp returned None type, the parser failed to parse the time
+        if not res:
+            await ctx.send(content='Failed to parse time duration. Please try again.')
             return None
 
-        padded_name = '\"' + name + (' ' if name else '') + '\"'
+        res = res[0]
+        time_diff = res[0] - curr_time
+        # Offset the lost second due to processing
+        sec_duration = time_diff.total_seconds() + 1
 
-        await ctx.send(content=f'<@{ctx.author.id}> The {padded_name}timer is set for {duration} {unit}(s)',
+        # Obtain the reminder content
+        timer_name = ''.join(message[i] for i in range(len(message)) if i < res[2] or i >= res[3]).strip()
+        if timer_name:
+            padded_name = '\"' + timer_name + '\" '
+        else:
+            padded_name = ''
+
+        # Format the output time string
+        days, remainder = divmod(sec_duration, 86400)
+        hours, remainder = divmod(remainder, 3600)
+        minutes, seconds = divmod(remainder, 60)
+
+        time_string = []
```

Comment:
From this to line 228 feels a bit verbose/long winded to me, however I'm not sure if it should be reimplemented as 
```Python
time_str = [(days, 'day'), (hours, 'hour'), (minutes, 'minute'), (seconds, 'second')]

# Function to handle converting the time unit into the correct string equivalent
ftime = lambda x, unit: f'{int(x)} {unit}{"s" if x > 1 else ""}'

# Only put in the time units that are more than 0 in our final formatted string
time_str = [ftime(*t) for t in time_str if t > 0]

# Add 'and' if the final time has more than 2 time units
time_str[-1] = f'and {time_str[-1]}' if len(time_str) > 2 else time_str[-1]

# Join string together
time_str = ', '.join(time_str)
``` 
because it isn't as readable. I'd like your input on this.